plugins/polio/js/src/domains/Calendar/campaignCalendar/map/CalendarMapPanesMerged.tsx

```diff
@@ -1,39 +1,46 @@
 /* eslint-disable camelcase */
-import React, { FunctionComponent } from 'react';
+import React, { FunctionComponent, useCallback } from 'react';
 import { GeoJSON, Pane } from 'react-leaflet';
 
+import { GeoJson } from '../../../../constants/types';
+import {
+    polioVaccines,
+    useOtherVaccineColor,
+} from '../../../../constants/virus';
+import { MergedShapeWithCacheDate } from '../types';
 import { CalendarMapTooltip } from './CalendarMapTooltip';
 import { getGeoJsonStyle } from './utils';
-import { polioVaccines } from '../../../../constants/virus';
-import { MergedShapeWithColor } from '../types';
 
 type Props = {
-    mergedShapes: MergedShapeWithColor[];
+    mergedShapes: MergedShapeWithCacheDate[];
     zoom: number;
 };
 
 export const CalendarMapPanesMerged: FunctionComponent<Props> = ({
     mergedShapes,
     zoom,
 }) => {
+    const otherVaccineColor = useOtherVaccineColor();
+
+    const getColor = useCallback(
+        (vaccine: string | undefined) => {
+            const vaccineColor = polioVaccines.find(
+                v => v.value === vaccine,
+            )?.color;
+            return vaccineColor || otherVaccineColor;
+        },
+        [otherVaccineColor],
+    );
+
```

Comment:
if `otherVaccineColor`was in a `const` we could extract `getColor` and skip `useCallback`